src/main/java/org/sonarlint/intellij/fix/ShowFixSuggestion.kt

```diff
@@ -39,22 +39,22 @@ import org.sonarsource.sonarlint.core.rpc.protocol.client.fix.FixSuggestionDto
 class ShowFixSuggestion(private val project: Project, private val file: VirtualFile, private val fixSuggestion: FixSuggestionDto) {
 
     fun show() {
-        runOnUiThread(project, ModalityState.defaultModalityState()) {
-            val fileEditorManager = FileEditorManager.getInstance(project)
-            val psiFile = PsiManager.getInstance(project).findFile(file) ?: return@runOnUiThread
-            val document = file.getDocument() ?: return@runOnUiThread
-
-            if (!isWithinBounds(document)) {
-                get(project).simpleNotification(
-                    null,
-                    "Unable to open the fix suggestion, your file has probably changed",
-                    NotificationType.WARNING
-                )
-                return@runOnUiThread
-            }
+        val fileEditorManager = FileEditorManager.getInstance(project)
+        val psiFile = PsiManager.getInstance(project).findFile(file) ?: return
+        val document = file.getDocument() ?: return
```

Comment:
`getDocument` should be run in a read action, either keep `runOnUiThread` or `runReadActionSafely`